### ISSUE TITLE
Haxe development won't retry running flash tests and therefore hangs

### ISSUE THREAD

**Issue by derRaab** [derRaab/de-superclass-haxe#3]
Haxe development fails running flash tests - well - at least it won't retry:
https://travis-ci.org/derRaab/de-superclass-haxe/jobs/461261375

But with Haxe 3.4.7 everything seems to be fine:
https://travis-ci.org/derRaab/de-superclass-haxe/jobs/461261371

I tried to figure out what's happening but I don't get why this runs:
https://travis-ci.org/andyli/HaxeCI/jobs/461105760

**Comment by derRaab** [comment]
According to this https://twolfson.com/2013-09-22-debugging-travis-ci the problem has nothing to do with the `Xlib: extension "RANDR" missing on display ":99.0" error`.

**Comment by andyli** [comment]
That looks like Flash Player displayed an run-time error, which the FP test script was incapable of catching/displaying it to the build log. Have you tried testing the Haxe dev build locally on your machine?

**Comment by derRaab** [comment]
@andyli - actually I'm not running the flash test locally since I always get this Flash Player Security Error:

```
SecurityError: Error #2017: Only trusted local files may cause the Flash Player to exit.
	at flash.system::System$/exit()
	at TestMainHx$/main()
	at boot_2723/init()
	at flash::Boot/start()
	at boot_2723()
```

And this is (to my understanding) an expected behavior so I actually don't know why it works at all in the ci environment. I'm confused. :)

**Comment by andyli** [comment]
Yeah, that particular error is expected. You can either comment out the `System.exit()` when testing it locally or just make the directory as "trusted". In CI, it works because we do the latter in the install flash player step:

https://github.com/derRaab/de-superclass-haxe/blob/f925028609a6ae160bffc8d997e2c7fae54d8fb8/dev/flash/Install.hx#L81-L83

Given the haxe 3.4 build passed, I believe the Haxe dev build gave you another error.

**Comment by derRaab** [comment]
Thanks for clarifying this. Stupid me, I overlooked these lines. Local runs just fine now.
Do you know who's is responsible for the retry handling:

```
$ haxe dev/ci/build-flash.hxml -D fdb && haxe dev/flash/run.hxml bin/flash/TestMainHx.swf
Xlib:  extension "RANDR" missing on display ":99".
Application crashed with an unhandled SIGSEGV
retry... (1)
```

Because this is basically what's missing in failing build.

**Comment by andyli** [comment]
The retry thing is at
https://github.com/derRaab/de-superclass-haxe/blob/f925028609a6ae160bffc8d997e2c7fae54d8fb8/dev/flash/Run.hx#L23-L32
